cli/src/main/java/org/jboss/as/cli/handlers/BaseOperationCommand.java

```diff
@@ -265,6 +266,9 @@ protected void addHeaders(CommandContext ctx, ModelNode request) throws CommandF
             return;
         }
         final ModelNode headersNode = headers.toModelNode(ctx);
+        if (headersNode != null && headersNode.getType() != ModelType.OBJECT) {
+            throw new IllegalArgumentException("--headers option has wrong value '"+headersNode+"'");
```

Comment:
The words in the class name do in fact describe the case but the CLI shouldn't throw runtime exceptions when parsing or handling command lines. If the command is wrong it has to be CFE.